### PR TITLE
Refactor image upload requirements as a checker

### PR DESCRIPTION
**Pull request by benthorner**
https://trello.com/c/4iKDzhBY/379-separate-saving-a-change-from-the-creation-of-a-preview

This refactors the code to compute issues for image uploads so that it
matches the checker approach we use everywhere else. Part of this
approach is that we only check for one or two issues in the feature
specs, and use a unit spec to cover all of them in one place.

### CODE REVIEW

File: app/services/requirements/image_upload_checker.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+    def issues
+      issues = []
+
+      unless file
+        issues << Issue.new(:image_upload, :no_file)
+        return CheckerIssues.new(issues)
+      end
+
+      if allowed_format? || animated_image?
```
Comment (1pretz1) [inline app/services/requirements/image_upload_checker.rb]: I think this should be `disallowed_format?` or equivalent

File: app/services/requirements/image_upload_checker.rb [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    def issues
+      issues = []
+
+      unless file
+        issues << Issue.new(:image_upload, :no_file)
+        return CheckerIssues.new(issues)
+      end
+
+      if unsupported_type? || animated_image?
+        issues << Issue.new(:image_upload, :format_not_allowed)
```
Comment (kevindew) [inline app/services/requirements/image_upload_checker.rb]: One of the things I'd suggest we do (but not a priority) is to split up type and animated into separate checks. I'd suggest the static gif terminology of the message is a bit confusing and it'd be easier to just have a separate message that was "You must... Upload an animate that is not animated"

Comment (benthorner) [reply app/services/requirements/image_upload_checker.rb]: Yeah, I was wondering about that - wanted it to be a pure refactor, though ;-). I've made a note and might get on to it later.

Comment (kevindew) [reply app/services/requirements/image_upload_checker.rb]: Yup, no problem - I missed that first time round so thought I'd drop it in.

**Pull request merged**